modules/data.remote/R/download.thredds.R

```diff
@@ -20,84 +30,124 @@
 ##' @export
 ##' @author Bailey Morrison
 ##'
-download.thredds.AGB <- function(outdir = NULL, site_ids, run_parallel = FALSE, 
-                                    ncores = NULL) {
+download.thredds.data <- function(outdir = NULL, site_info, dates = c("19950201", "19961215"), 
+                                  varid = "LAI",
+                                  dir_url = "https://www.ncei.noaa.gov/thredds/catalog/cdr/lai/files", 
+                                  data_url = "https://www.ncei.noaa.gov/thredds/dodsC/cdr/lai/files",
+                                  run_parallel = TRUE) {
+  # require("XML")
+  # require("RCurl") 
+  require("foreach")
+  
+  # check that dates are within the date range of the dataset
+  dates = c(as.Date(dates[1], "%Y%m%d"), as.Date(dates[2], "%Y%m%d"))
+  if (!(is.null(dir_url)))
+  {
+    #https://www.ncei.noaa.gov/thredds/catalog/cdr/lai/files/1981/catalog.html -> link for directory files, not downloads
+    result <- RCurl::getURL(paste(dir_url, "catalog.html", sep = "/"), verbose=F,ftp.use.epsv=TRUE, dirlistonly = TRUE)
+    files = XML::getHTMLLinks(result)
+    
+    date_year_range = unique(range(c(year(as.Date(dates[1], "%Y")), year(as.Date(dates[2], "%Y")))))
```

Comment:
`as.Date` on a Date object and `range` on two values are both no-ops, so this can be simplified to:
```suggestion
    date_year_range = unique(lubridate::year(dates))
```